### PR TITLE
cli: Various improvements

### PR DESCRIPTION
**Pull request by vadorovsky**
* Download circom as one of binaries managed by us. That ensures that it's always installed. And only we provide a macos-arm64 binary (upstream doesn't).
* Provide a new `executeCommand()` function, which in contrast to `execSync()`, outputs the stdout and stderr of a child process in live manner. On success, the accumulated stdout is also returned as a string.

### CODE REVIEW

**Comment by ananas-block** [comment]
when trying to do yarn install in the cli folder I got the following error:
```
error An unexpected error occurred: "https://registry.yarnpkg.com/cli-progress/-/cli-progress-3.12.0.tgz: getaddrinfo EAI_AGAIN registry.yarnpkg.com".
```

I resolved this error by adding cli-progress as a regular dependecy, so far it is only a dev dependency:
```
yarn add cli-progress
```

**Comment by ananas-block** [comment]
did you add the stdout streaming to light test as well?
because my output still only appeared once the test passed

**Comment by ananas-block** [comment]
apart from the two comments above it worked for me :)

**Comment by ananas-block** [comment]
also I am getting this weird print during testing:
image

**Comment by ananas-block** [comment]
> > when trying to do yarn install in the cli folder I got the following error:
> > ```
> > error An unexpected error occurred: "https://registry.yarnpkg.com/cli-progress/-/cli-progress-3.12.0.tgz: getaddrinfo EAI_AGAIN registry.yarnpkg.com".
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > I resolved this error by adding cli-progress as a regular dependecy, so far it is only a dev dependency:
> > ```
> > yarn add cli-progress
> > ```
> 
> cli-progress is already added as a normal dependency here https://github.com/Lightprotocol/light-protocol/pull/137/files#diff-088b5e0ac52dd36816ebc6b121d5423c763c3c18c098b91757c014937c7d632dR27
> 
> Your error probably means that you had some DNS or connectivity issue. Does it occur constantly? I didn't see it.
> 
> Only `@types/cli-progress` is in devDependencies, but that's alright, because it's only for Typescript, needed for compilation.

weird maybe I fucked up the install, and didn't do a force install, just removed the node modules and installed. Maybe that was not enough.

File: cli/src/commands/test.ts [hunk 1]
```diff
@@ -33,10 +33,16 @@ export const handler = async (argv: Arguments<Options>): Promise<void> => {
...
-    );
-    console.log(stdout.toString().trim());
+    executeCommand({
+      command: commandPath,
+      args: [
+        systemProgramPath,
+        process.cwd(),
+        programAddress,
+        `${programName}.so`,
+        `'yarn ts-mocha 2000000 tests/${projectName}.ts --exit'`
```
Comment (ananas-block) [inline cli/src/commands/test.ts:43]: I think the -t flag is missing before 2000000
'yarn ts-mocha -t 2000000 tests/${projectName}.ts --exit'

**Review by SwenSchaeferjohann — APPROVED** [review]
lgtm

**Pull request merged**